src/impl/apis/data_api_impl.py

```diff
@@ -258,35 +261,167 @@ async def get_time_range(
+            elif is_flow:
+                data_by_date[this_date].setdefault(row[origin_index], {})
+                if value is not None:
+                    data_by_date[this_date][row[origin_index]][row[destination_index]] = value
+            else:
+                raise HTTPException(
+                    status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+                    detail="Cannot find spatial unit in data",
+                )
+
+    logger.debug(
+        "Finished running data query",
+        num_results=num_rows,
+        min_value=min_value,
+        max_value=max_value,
+    )
+    if len(data_by_date.keys()) == 0:
+        raise HTTPException(status_code=HTTPStatus.NOT_FOUND, detail="No data found")
+
+    new_result = QueryResult(
+        min=min_value if min_value != math.inf else None,
+        max=max_value if max_value != -math.inf else None,
+        data_by_date=dict(sorted(data_by_date.items())),
+    )
+
+    return new_result
+
+
+async def get_column_names(table_name: str, metadata: List[Metadata], pool: Pool) -> List:
+    select_query = (
+        f"SELECT * FROM `{os.getenv('DB_NAME')}`.`{table_name}_{metadata[0].mdid}` LIMIT 1;"
+    )
+    logger.debug("Getting column names", table_name=table_name)
+    async with pool.acquire() as conn, conn.cursor() as cursor:
+        await cursor.execute(select_query)
+        return [i[0] for i in cursor.description]
+
+
+async def stream_query(
+    base_table_name: str, metadata: List[Metadata], pool: Pool, table_name: str
+) -> AsyncGenerator[list, None]:
+    table_names = [f"`{os.getenv('DB_NAME')}`.`{table_name}_{m.mdid}`" for m in metadata]
+    short_names = [table_name.rpartition(".")[2].strip("`") for table_name in table_names]
+
+    partition_queries = [
+        f"SELECT `{short_name}`.*, `dt` FROM {table_name} AS `{short_name}` LEFT JOIN metadata USING (mdid)"
+        for table_name, short_name in zip(table_names, short_names)
+    ]
+    select_query = " UNION ".join(partition_queries)
+
+    logger.debug(
+        "Running data query...",
+        base_table_name=base_table_name,
+        table_name=f"{table_name}_<mdid>",
+    )
+    async with pool.acquire() as conn, conn.cursor(SSCursor) as cursor:
+        await cursor.execute(select_query)
+
+        column_names = [i[0] for i in cursor.description]
+        logger.debug("Executed query", column_names=column_names)
+        if "data" not in column_names:
+            raise HTTPException(
+                status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+                detail="Cannot find data for metadata",
+            )
+        sleep_counter = 50
+        while True:
+            chunk = await cursor.fetchmany(CSV_CHUNK_SIZE)
+            sleep_counter -= 1
+            if sleep_counter == 0:
+                sleep_counter = 50
+                logger.debug("Sleeping query coroutine")
+                await sleep(10)
+            if chunk:
```

Comment:
This will work, but is generally bad practice to take advantage of the truthy properties of lists, better to check if the list is empty and break if it is I think.